### DISCUSSION TITLE
ffmpegthumbnailer support in Fedora

### DISCUSSION THREAD

**Discussion by mke1976** [gerbera/gerbera#1234] [Q&A] [answered]
Currently the gerbera ffmpegthumbnailer support is turned off by compiler switch in the Fedora rpm.

I have raised a ticket, asking the maintainer to change this: https://bugzilla.redhat.com/show_bug.cgi?id=1928109

The issue with the request is, that ffmpegthumbnailer is not in the official Fedora repo _(it is in the rpmfusion repo)_. Therefore  ffmpegthumbnailer cannot be linked as a dependency in the gerbera rpm and it cannot be guranteed that ffmpegthumbnailer is really installed on the system.

Now my questions: Assuming the gerbera ffmpegthumbnailer compiler switch is turned on, but ffmpegthumbnailer is not installed, what would happen?

1. The default (Fedora) /etc/gerbera/config.xml is used. It doesn't mention ffmpegthumbnailer at all. _(My guess: Gerbera uses **ffmpegthumbnailer enabled="no"** and the program will run as per normal. Is this correct?)_

2. The user has changed the setting to **ffmpegthumbnailer enabled="yes"**, but ffmpegthumbnailer is not installed. Will gerbera crash or will it just ignore the setting?

Thank you for your answers in advance!

**Comment by whyman** [comment, marked as answer]
Gerbera will not start if the library is missing, as the linker will complain.

**Reply by mke1976** [reply]
Thank you for the quick feedback!

That means that distributions that don't ship ffmpegthumbnailer out of the box cannot support the thumbnail feature.

Just an idea for future versions: Would it be possible to test for the external library and only enable a function if it is present?

**Reply by whyman** [reply]
We could maybe do something like dlopen-ing the library if available, but someone would need to contribute a patch for that.

I dont see why fedora would have issues with the thumbnailer package though, (assuming they ship ffmpeg itself) as it has no codecs or anything in it, it just uses what is available in ffmpeg :shrug:

**Comment by mke1976** [comment]
I just checked. ffmpeg is not in the official Fedora repo. You need to enable the 3rd party rpmfusion repos.
